### ISSUE TITLE
Black overlay on Ubuntu 20.04

### ISSUE THREAD

**Issue by sfjuocekr** [trigg/DiscordOverlayLinux#60]
I just get a black overlay on KUbuntu 20.04 LTS.

There should probably be an easier way to get back to the configuration screen or maybe there is and I haven't found it!

**Comment by ls-1N** [comment]
You can get back to the config screen by right-clicking the tray icon and choosing "Settings". In addition to the Discord tray icon (which you might have enabled or disabled), discord-overlay also has a tray icon.

Try running the app from terminal with `discord-overlay` to see if there are any errors. Then websearch any errors or post them here formatted into a code block \`\`\`. If it is really long, put it between \<details\> \<\/details\>

Also maybe instructions of what worked for me on Kubuntu 16.04 can help you: #61

**Comment by midget3111** [comment]
If the overlay is black, you need to enable Compositing :)

**Comment by sfjuocekr** [comment]
Yea, yesterday I figured out that the black was probably caused by compositing being off. Anyway, I turned it on and now I have a transparent overlay. Normally I only have XRender as the backend, but I tried OpenGL 2.0 and 3.1.

I never noticed the tray icon, my bad :D

Either way, the overlay is now just transparent and when I set it up the preview works but the overlay does not show anything.

libxcb-xinerama1 is already installed and this is the output (albeit, not very useful):

`2020-09-25 14:05:46,703 DEBUG    Discovered screens: ['HDMI-0', 'DVI-D-0']
2020-09-25 14:05:46,703 DEBUG    Chose screen HDMI-0
js: [object Object]`

I've also tried with QT_SCREEN_SCALE_FACTORS=1 instead of 2 (I use a 4K monitor) to see if scaling had something todo with it.

**Comment by trigg** [comment]
If the preview works but nothing appears on the overlay, go back into settings, set up the preview again and press 'Use this room' or 'Use all rooms'

After that double check positioning, but it sounds like that's not the issue.

**Comment by ls-1N** [comment]
Yeah configuration isn't **the most** intuitive possible (but it's good, I guess), but you should start seeing something on the screen if you configure it correctly. Fully transparent seems like the software is working properly.

**Comment by sfjuocekr** [comment]
I've tried to remake the overlay a bunch of times and also wiped ~/.config/discord-overlay to start fresh.

Can't get it to show the overlay even if I make it span the entire screen!

When I open the overlay, it briefly shows the black overlay before going transparent.

**Comment by ls-1N** [comment]
For me the out of the box instructions for using it for voicechat:
1. launch Discord
2. join any voice channel. (You can mute yourself).
3. launch discord-overlay
4. (open Settings, if not open yet)
5. click Layout
6. click Voice Widget
7. click Select a Server
8. choose the server you are on
9. click Voice Channel
10. choose the channel you are on
11. change any settings you want here
12. click Use this room

And your're basically done. You may close the settings window.
13. adjust Position, if you want. (The sliders are min and max area).

**Comment by sfjuocekr** [comment]
I've tried exactly this numerous times, I even tried all three different compositor settings and unchecked the box to allow the compositor to be turned off:

https://youtu.be/qHmLydQSYMQ

**Comment by sfjuocekr** [comment]
Ok, maybe I had to be really patient and it just started working magically!

I was just toying around with the sliders, trying to switch monitors and it just appeared and now it works!

**Comment by sfjuocekr** [comment]
Oh nevermind, I figured it out!

The option "hideinactive" set to 1 just doesnt show when there is a single person in the room I guess? Anyway, set to 0 and it work when solo testing!

**Comment by ls-1N** [comment]
Interesting. If this is the same as "Hide voice channel when inactive" in GUI, then I have it enabled and it worked by myself fine for me. But as long as it is solved, it doesn't matter much
Also they made a [new one](https://github.com/trigg/Discover) of this. Btw your video is private.

**Comment by sfjuocekr** [comment]
Ah, I fixed the video and made it unlisted!

But yea it is the "hide voice channel when inactive" it just seems to hide everything when there is just one person in the channel for me.

**Comment by yalin** [comment]
It does not show at all. I'm using 2 monitors, it may be the problem?

**Comment by sfjuocekr** [comment]
Im using two monitors as well, you need to swap the output once for it to pickup!

The black is caused by the compositor being disabled, Proton likes to force it off so if you have to disallow the compositor being turned off. Which obviously brings big performance implications for some games, for example I can't even leave my browser open playing a video or the FPS drops to below 60 while streaming.

I think, if we can get this into a Vulkan layer... we should be golden!

Also, I'd like to see KapChat fuctionallity. You can just change the URL in the configfile to have discord + twitch overlays.
